frontend/src/tests/AuthGuard.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { render, screen } from '@testing-library/react'
+import { AuthContext, AuthStatus } from '../components/useAuthContext'
+import { AuthGuard } from '../components/AuthGuard'
+import { expect } from 'vitest'
+import { PageContextProvider } from '../components/usePageContext'
+import { type PageContext } from 'vike/types'
+
+describe('AuthGuard', () => {
+  it('Authguard denies user protected content if user is not logged in', () => {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const pageContext = {
+      urlPathname: '/about'
+    } as PageContext
+
```

Comment:
I noticed that this test causes `jsdom` to emit `Error: Not implemented: navigation (except hash changes)`. The test itself seems to work, so perhaps this is benign, but thought I'd note it just in case there's some way to clean that up :man_shrugging: .